src/main/java/com/mucommander/main/AutoProcessor.java

```diff
@@ -219,20 +255,36 @@ else if ((b != null) && actionList.contains(AUTO_DEPLOY_UPDATE_VALUE))
                 }
             }
 
+
             // Start all installed and/or updated bundles if the 'start'
             // action is present.
             if (actionList.contains(AUTO_DEPLOY_START_VALUE))
             {
-                for (int i = 0; i < startBundleList.size(); i++)
+                for (Future<Bundle> bundleFuture : startBundleList)
                 {
-                    try
-                    {
-                        ((Bundle) startBundleList.get(i)).start();
-                    }
-                    catch (BundleException ex)
-                    {
-                        System.err.println("Auto-deploy start: "
-                            + ex + ((ex.getCause() != null) ? " - " + ex.getCause() : ""));
+                    try {
+                        long pre = System.currentTimeMillis();
+                        var bundle = bundleFuture.get();
+                        System.out.println("------ get bundle " + (System.currentTimeMillis() - pre) + "ms " + bundle);
+                        var startLvl = startLevel;
+                        executor.execute(() -> {
+                            try
+                            {
+                                if (bundle != null) {
+                                    if (!isFragment(bundle)) {
+                                        sl.setBundleStartLevel(bundle, startLvl);
+                                    }
+                                    bundle.start();
```

Comment:
previously, we wouldn't have started a bundle for which `isFragment(bundle) == true`, did you intentionally change this?

Reply:
I guess my mistake. I will move .start it into if-clause